### PR TITLE
Clarify what a "module" is and update feature profiles for clients

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/matrix-org/matrix-doc/issues/2009

**Note**: There's room for bikeshed here. Please keep it to a minimum/none at all.

### CODE REVIEW

File: specification/modules.rst [hunk 1]
```diff
@@ -15,3 +16,10 @@
 Modules
 =======
 
+Modules are parts of the Client-Server API which are not universal to all
+endpoints and are accessible to all clients. Modules are strictly defined
```
Comment (ara4n) [inline specification/modules.rst]: "and are accessible" - unsure what you're getting at here?

Comment (turt2live) [reply specification/modules.rst]: not a clue. I'll chop that part off.

File: specification/modules.rst [hunk 2]
```diff
@@ -15,3 +16,10 @@
...
+within this specification and should not be mistaken for XEP or equivalent
```
Comment (ara4n) [inline specification/modules.rst]: let's not define ourselves in terms of XEPs or XMPP.  perhaps: "...should not be mistaken for experimental extensions or optional features.  A compliant server implementation MUST support all modules and supporting specification (unless the implementation only targets clients of certain profiles, in which case only the modules for those feature profiles MUST be implemented).  A compliant client implementation MUST support all the modules and supporting specification for the Feature Profile it targets."

I think we should make it easy for people to implement compliant but partial servers (e.g. dendrite) if they know the server is only intended to target a limited client profile.

File: specification/feature_profiles.rst [hunk 3]
```diff
@@ -32,33 +33,67 @@ Summary
   Module / Profile                       Web       Mobile    Desktop       CLI     Embedded
 ===================================== ========== ========== ========== ========== ==========
  `Instant Messaging`_                  Required   Required   Required   Required   Optional
+ `Direct Messaging`_                   Required   Required   Required   Required   Optional
```
Comment (ara4n) [inline specification/feature_profiles.rst]: these all feel fine ftr.

**Comment by turt2live** [comment]
plan is to merge this on Monday morning to give people time to comment btw

**Comment by turt2live** [comment]
it seems like this is not very controversial, so merging.

**Pull request merged**